treatment-service/config/config.go

```diff
@@ -88,10 +88,28 @@ type Monitoring struct {
 	MetricLabels []string       `json:"metric_labels" default:""`
 }
 
+// MessageQueueKind describes the message queue for transmitting event updates to and fro Treatment Service
+type MessageQueueKind = string
+
+const (
+	// NoopMQ is a No-Op Message Queue
+	NoopMQ MessageQueueKind = ""
+	// PubSubMQ is a PubSub Message Queue
+	PubSubMQ MessageQueueKind = "pubsub"
+)
+
+type MessageQueueConfig struct {
+	// The type of Message Queue for event updates
+	Kind MessageQueueKind `default:""`
+
+	// PubSubConfig captures the config related to subscribing to a PubSub Message Queue
+	PubSubConfig PubSub
+}
+
 type PubSub struct {
-	Project              string `json:"project" default:"dev" validate:"required"`
-	TopicName            string `json:"topic_name" default:"xp-update" validate:"required"`
-	PubSubTimeoutSeconds int    `json:"pub_sub_timeout_seconds" default:"30" validate:"required"`
+	Project              string `json:"project" default:"dev"`
+	TopicName            string `json:"topic_name" default:"xp-update"`
+	PubSubTimeoutSeconds int    `json:"pub_sub_timeout_seconds" default:"30"`
```

Comment:
Why are we removing the validation tags? Can we have `MessageQueueConfig` have a pointer to the `PubSub`object instead so that field is nullable? i.e.
```go
type MessageQueueConfig struct {
	Kind MessageQueueKind `default:""`
	PubSubConfig *PubSub
}
```

Reply:
👍🏻 reverted!